### PR TITLE
Javadoc for `Aggregator#getLeafCollector`

### PR DESCRIPTION
**Pull request by nik9000**
This adds some javadoc for `Aggregator#getLeafCollector` and some
related classes.

### CODE REVIEW

**Review by benwtrent — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/search/aggregations/LeafBucketCollector.java [hunk 1]
```diff
@@ -36,36 +42,6 @@ public boolean isNoop() {
         }
     };
 
-    public static LeafBucketCollector wrap(Iterable<LeafBucketCollector> collectors) {
```
Comment (not-napoleon) [inline server/src/main/java/org/elasticsearch/search/aggregations/LeafBucketCollector.java:39]: Was this just not being used?  Seems a little surprising to be removing it in a docs change.

**Review by not-napoleon — APPROVED** [review]
This looks really good.  I think it's very confusing how we use `NO_OP_COLLECTOR`, and this helps clarify that.

File: server/src/main/java/org/elasticsearch/search/aggregations/LeafBucketCollector.java (hunk 1 above)
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/search/aggregations/LeafBucketCollector.java:39]: Sorry. Yeah, it was not being used at all. I can slice out the change though.

**Comment by nik9000** [comment]
Thanks for looking @benwtrent and @not-napoleon!

**Pull request merged**